.github/workflows/publish-winget.yml

```diff
@@ -0,0 +1,74 @@
+# Create winget pacakge YAML file and open a PR in the repository
+# 
+# Required secrets:
+# - WINGET_UPSTREAM_REPO - repository with the YAML files, by default microsoft/winget-pkgs
+# - WINGET_WORKER_NAME - GitHub username of the machine account that will create the PR. The account
+#                        must have a fork of the upstream repo
+# - WINGET_WORKER_EMAIL - machine account e-mail
+# - WINGET_WORKER_PAT- machine account Personal Access Token
+
+name: publish-winget
+
+on:
+  release:
+    types: [published]
```

Comment:
Just a quick note on expectations: I'm pretty sure this will only land if the published tag includes this file, which means once it lands we should backport it to every active major.

Reply:
This file only needs to be in master, it will work for all published tags. The script also works if it is called from other repos (like [this PR](https://github.com/microsoft/winget-pkgs/pull/2226) made from [this release in my fork](https://github.com/bzoz/node/releases/tag/v12.18.2))

Reply:
Interesting, that's quite inconsistent with how other Actions events work, but if it works it works :)

Reply:
You've got a point there, I'll check it out. I've written the script some time ago, might as well check if it still works.

Reply:
I'm checking on a personal repository, will share the results soon

Reply:
It only runs if the action exists in the release, I pushed [a simple action](https://github.com/mmarchini-oss/automated-merge-test/blob/master/.github/workflows/onpublish.yml) that runs on publish, then I published 0.0.2 from master, which [triggered the run](https://github.com/mmarchini-oss/automated-merge-test/actions?query=workflow%3Arun-on-publish), and then I published 0.0.1 from the commit right before the action was added, which [didn't trigger the run](https://github.com/mmarchini-oss/automated-merge-test/actions?query=workflow%3Arun-on-publish).

Actions have weird semantics sometimes.

Reply:
Huh, good to know. Thanks for investigating this.

Reply:
No problem. Also, this doesn't block or affect this PR, it's just a heads up of what we need to do once we land it.